lua/possession/query.lua

```diff
@@ -6,9 +6,11 @@ local config = require('possession.config')
 
 --- Get a sessions as a list-like table
 ---@param sessions? table<string, table> like from possession.session.list()
+---@param only_cwd? boolean only load sessions for the cwd
 ---@return table[] list of session data with additional `file` key
-function M.as_list(sessions)
-    sessions = sessions or session.list() --[[@as table<string, table> ]]
+function M.as_list(sessions, only_cwd)
```

Comment:
I think we can instead provide a separate function `query.filter(sessions, opts)` that would take the output of `query.as_list` and `opts` for filtering (for now it could only support `opts.cwd`). `opts.cwd` would be `true` to use `vim.fn.getcwd()`  and (optional feature) could be a string to use the given cwd. 

The implementation would probably just be `return vim.tbl_filter(function(s) ... end, sessions)`, so maybe it would make sense just return this `function(s) ... end`  (an maybe name it `query.make_filter`)? But that's probably over-engineering :P But I'd vote for having `query.filter` as separate function.

Reply:
All good points, the structure of the different modules is starting to make more sense now, thanks. I've made these changes. `query.filter` is limited to just `cwd` intentionally for now. `cwd` is just a string, so the value of it can be determined by the caller to the filter function, which is preferable to `filter` making this choice.

The filter function could be generalised I think with a `for ... pairs(opts)` filtering on `sessions[opts.key] == opts.value` (pseudo code) if needed at a later time.

I'm not familiar with how Neovim manages whatever different working directories it might have, so please check that `commands.get_list` is using the correct one.

Reply:
Yeah, you're right, it's better to have cwd only as string and leave it up to the caller.

As for the `vim.fn.getcwd` calls I think now they are correct. Most users don't use tab/window-local cwd, but I personally often use tab-local cwd. This way I can have tabs open "per-project" and e.g. `Telescope files` from each shows only files from this project. You can check the documentation of `:cd`/`:tcd`/`:lcd`, but basically they have priority: window-local > tab-local > global, and if *-local is not set it defaults to the one with lower priority.